spec/models/user_group_spec.rb

```diff
@@ -1,22 +1,67 @@
 require 'rails_helper'
 
 RSpec.describe UserGroup, :type => :model do
-  context 'with groceries' do
+  describe 'group groceries' do
+    context 'with groceries' do
+      let(:user_group) { create(:user_group, :with_groceries) }
 
-    it 'scopes by unfinished' do
-      group = create(:user_group, :with_groceries)
-      result = group.active_groceries
+      it 'scopes by unfinished' do
+        result = user_group.active_groceries
+        expect(result).to eq user_group.groceries
+      end
 
-      expect(result).to eq group.groceries
+      it 'scopes by finished' do
+        user_group.groceries.update_all(finished_at: DateTime.now)
+        result = user_group.finished_groceries
+        expect(result).to eq user_group.groceries
+      end
+    end
+  end
+
+  describe 'privacy scoping' do
+    before(:each) do
+      @public_group = create(:user_group, :with_groceries, privacy: UserGroup::PUBLIC)
+      @public_group2 = create(:user_group, :with_groceries, privacy: UserGroup::PUBLIC)
+      @private_group = create(:user_group, :with_groceries, privacy: UserGroup::PRIVATE)
+    end
+
+    context 'public group' do
+      it 'should filter by public user groups' do
+        expect(UserGroup.public.length).to eq 2
+      end
+
+      it 'should return all public items for a public group' do
+        items = @public_group.items.merge(@public_group2.items)
+        expect(@public_group.privacy_items).to eq items
+      end
     end
 
-    it 'scopes by finished' do
-      group = create(:user_group, :with_groceries)
-      group.groceries.update_all(finished_at: DateTime.now)
-      result = group.finished_groceries
+    context 'private group' do
+      it 'should filter by private user groups' do
+        expect(UserGroup.private.length).to eq 1
+      end
 
-      expect(result).to eq group.groceries
+      it 'should return all public items for a private group' do
+        expect(@private_group.privacy_items).to eq @private_group.items
+      end
+    end
+  end
+
+  describe 'invitations' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.